pyramda/prop.py

```diff
@@ -0,0 +1,8 @@
+from pyramda.function.curry import curry
+
+
+@curry
+def prop(x, obj):
+    if isinstance(obj, dict):
```

Comment:
I'm a little hesitant on this. Before I remove the dictionary part maybe you could elaborate a little, along the lines of why that would be problematic or not so useful, or even aesthetically undesirable etc?